lib/usagereporter/teleport/aggregating/reporter.go

```diff
@@ -211,10 +224,14 @@ func (r *Reporter) run(ctx context.Context) {
 
 	userActivity := make(map[string]*prehogv1.UserActivityRecord)
 
-	userRecord := func(userName string) *prehogv1.UserActivityRecord {
+	userRecord := func(userName string, v1AlphaUserKind prehogv1alpha.UserKind) *prehogv1.UserActivityRecord {
```

Comment:
Nit: I don't think it's a major concern but I can see a bit of potential raciness here if there's an issue with tracking bot status. The status of the user from the first event will set it for all future events. Perhaps we ought to put some warning or error in if the kind of the user changes ?

Reply:
I added a warning and had it override the flag. I think the behavior could go either way realistically - and given it's a bug regardless I'm not sure if I expect anything good to come from whichever behavior we use.

Reply:
Maybe we should leave the `record.UserKind` as is if `v1UserKind` is unspecified, just in case?

Reply:
I've tweaked the logic here. It now allows identifying a previously unspecified user, ignores unspecified updates, and warns about flipping between human/bot users.